src/webgpu/shader/execution/expression/unary/bitwise.spec.ts

```diff
@@ -0,0 +1,63 @@
+export const description = `
+Execution Tests for the bitwise binary expression operations
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { i32, scalarType, u32 } from '../../../../util/conversion.js';
+import { fullI32Range, fullU32Range } from '../../../../util/math.js';
+import { allInputSources, run } from '../expression.js';
+
+import { unary } from './unary.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('complement')
+  .specURL('https://www.w3.org/TR/WGSL/#bit-expr')
+  .desc(
+    `
+e1 | e2: T
+T is i32, u32, AbstractInt, vecN<i32>, or vecN<u32>, vecN<AbstractInt>
+
+Bitwise-or. Component-wise when T is a vector.
+`
+  )
+  .params(u =>
+    u
+      .combine('type', ['i32', 'u32'] as const) // TODO: Test AbstractInt
```

Comment:
This is where I put the TODO.
Let me know if there is a better pattern.

Reply:
For type parameterized tests like this, the pattern I have used is to break them up by type, so there would be a `g.test('complement_u32')`, `g.test('complement_i32')`, and `g.test('complement_abstractint'), etc. So you can use .unimplemented instead of .fn for the ones that are not implemented, and have the TODO ending up in the description to avoid the linter issue about the comment.

This also allows for avoiding doing a lot of the switching logic based on type that you have to do. Any really common patterns between tests, can still be pulled up into a helper function at the top of the file.